Add tests to check forwarding sysctls

sysvinit opened the pull request:
This change is a follow-up to #1351, which adds a NixOS to check if the default platform sysctl configuration correctly disables IP forwarding.

PL-133557

@flyingcircusio/release-managers

## Release process

- [x] ~~Created changelog entry using `./changelog.sh`~~ => not necessary


## PR release workflow (internal)

- [x] PR has internal ticket
- [x] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [x] ticket is on Platform agile board
- [x] ticket state set to *Pull request ready*
- [ ] if ticket is more urgent than within the next few days, directly contact a member of the Platform team

## Design notes

- [ ] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [ ] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - This is a regression test which should compliment a previous change.
- [x] Security requirements tested? (EVIDENCE)
  - Hydra tests.

sysvinit committed:
tests: disable automatic ip address assignment by the test harness

By default, the NixOS test harness will automatically assign IP
addresses to interfaces inside the test VMs based ont he VM's internal
ID number and the index of the VLAN the interface is connected to.

However, if we're building a test VM using testlib.fcConfig, then
we'll be providing ENC data with IP address configuration which will
be consumed by the platform to set IP adresses itself. Let's ask the
NixOS harness not to do this, we should only be configuring IP
addresses in one place.

sysvinit committed:
tests: add test to check correct default ip forwarding configuration

PL-133557

ctheune commented on tests/testlib.nix:
```diff
@@ -257,7 +257,7 @@ rec {
   fcVlanIfaces = mapAttrs' (vlan: vid: {
     name = "eth${vlan}";
-    value = { vlan = vid; assignIP = true; };
+    value = { vlan = vid; assignIP = false; };
```
Is this intentional? This looks like it might break a lot of tests and Hydra seems to agree.

sysvinit replied:
Yes, it was intentional. I wasn't sure if there were many tests which relied on the assumption that `assignIP = true` is set, and really we ought not to use that if we're providing our own IP address configuration via the testlib ENC – this results in VM interfaces being assigned *both* the addresses generated by the NixOS test framework and our own testing library. In the new forwarding test in this PR this led to some correctness issues, as some of the IP addresses were assigned to more than one test VM, which was quite confusing. I can `lib.mkForce` the setting in the forwarding test and leave the other tests as they were though.

sysvinit replied:
It turns out that `assignIP = true` also configures the /etc/hosts file, which is used for name resolution in a few tests. I've added extra handling for this and the tests which had problems now seem to pass.

sysvinit committed:
tests: add other nodes' srv addresses to the hosts file

This is required to ensure that file-based name resolution based on
/etc/hosts works correctly in the test environment. Normally this
would be handled by the test harness for interfaces whose IP addreses
are assigned automatically, but as we've disabled this functionality
we need to do the work for ourselves.
